pinot-common/src/main/java/org/apache/pinot/common/function/TransformFunctionType.java

```diff
@@ -113,7 +113,9 @@ public enum TransformFunctionType {
           List.of(SqlTypeFamily.CHARACTER, SqlTypeFamily.CHARACTER, SqlTypeFamily.CHARACTER, SqlTypeFamily.CHARACTER,
               SqlTypeFamily.CHARACTER), i -> i > 2)),
   JSON_EXTRACT_KEY("jsonExtractKey", ReturnTypes.TO_ARRAY,
-      OperandTypes.family(List.of(SqlTypeFamily.CHARACTER, SqlTypeFamily.CHARACTER))),
+      OperandTypes.family(
+          List.of(SqlTypeFamily.CHARACTER, SqlTypeFamily.CHARACTER, SqlTypeFamily.CHARACTER),
+          i -> i >= 2 && i <= 3)),
```

Comment:
(minor) No need to set upper bound
```suggestion
          i -> i > 1)),
```